libyul/backends/evm/EVMCodeTransform.cpp

```diff
@@ -498,7 +602,10 @@ void CodeTransform::generateMultiAssignment(vector<Identifier> const& _variableN
 {
 	solAssert(m_scope, "");
 	for (auto const& variableName: _variableNames | boost::adaptors::reversed)
+	{
 		generateAssignment(variableName);
+		decreaseReference(variableName.name);
```

Comment:
This is interesting, because if the reference counter reaches zero here, we do not actually even have to perform the assignment. Could you perhaps move the counter decrement inside the `generateAssignment` function and skip the actual assigning in there if it reaches zero?